python/lsst/ip/isr/calibType.py

```diff
@@ -0,0 +1,630 @@
+        self._metadata["DETECTOR_SERIAL"] = self._detectorSerial
+
+        if setDate:
+            date = datetime.datetime.now()
+            mdSupplemental['CALIBDATE'] = date.isoformat()
+            mdSupplemental['CALIB_CREATION_DATE'] = date.date().isoformat()
+            mdSupplemental['CALIB_CREATION_TIME'] = date.time().isoformat()
+
+        for key, value in kwargs.items():
+            mdSupplemental[key] = value
+
+        mdOriginal.update(mdSupplemental)
+
+    @classmethod
+    def readText(cls, filename):
+        """Read calibration representation from a yaml/ecsv file.
+
+        Parameters
+        ----------
+        filename : `str`
+            Name of the file containing the calibration definition.
+
+        Returns
+        -------
+        calib : `~lsst.ip.isr.IsrCalibType`
+            Calibration class.
+        """
+        data = ''
+        try:
+            data = Table.read(filename, format='ascii.ecsv')
+        except Exception as e1:
+            try:
+                with open(filename, 'r') as f:
+                    data = yaml.load(f, Loader=yaml.CLoader)
+            except Exception as e2:
+                raise e2(f"Failed to read as ecsv as well: {e1}")
+
+        if isinstance(data, dict):
+            return cls().fromDict(data)
+        else:
+            schema = afwTable.Schema()
+            keyDict = dict()
+            for colName in data.columns:
+                try:
+                    keyDict[colName] = schema.addField(colName,
+                                                       type=data[colName].dtype.type)
+                except KeyError:
+                    keyDict[colName] = schema.addField(colName, type="String", size=32)
+
+            table = afwTable.BaseCatalog(schema)
+            table.resize(len(data))
+
+            for colName in data.columns:
+                key = keyDict[colName]
+                if key.getTypeString() == 'String':
+                    for rowNum, row in enumerate(data[colName]):
+                        table[rowNum][key] = row
+                else:
+                    table[key] = data[colName]
+
+            metadata = PropertyList()
+            for k, v in data.meta.items():
+                metadata[k] = v
+            table.setMetadata(metadata)
+
+            return cls().fromTable([table])
+
+    def writeText(self, filename):
+        """Write the calibration data to a text file.
+
+        Parameters
+        ----------
+        filename : `str`
+            Name of the file to write.
+
+        Returns
+        -------
+        used : `str`
+            The name of the file used to write the data.
+
+        Raises
+        ------
+        RuntimeError :
+            Raised if filename does not end in a known extension
+
+        Notes
+        -----
+        The file is written to YAML/ECSV format and will include any
+        associated metadata.
+        """
+        outDict = self.toDict()
```

Comment:
I think this is only used in the yaml branch. It's also not implemented by default so will raise an exception even if ecsv is requested.